03/Product.js

```diff
@@ -1,43 +1,36 @@
 import React from "react";
 class Product extends React.Component {
-  state = {
-    cartArray: [],
-    productObj: {},
-  };
-
-  addToCart = (e, productId, productName, productPrice) => {
-    e.target.disabled = true;
+  addToCart = (productId, productName, productPrice) => {
     const productObj = {
         id: productId,
         name: productName,
         price: productPrice
     }
-    this.setState((prevState) => ({
-        cartArray: [...prevState.cartArray, { ...productObj }],
-      }));
-      this.props.addToCartHandler({ ...productObj });
-      
+      this.props.addToCartHandler({ ...productObj }); 
   };
-  removeFromCart = (id) => {
-    this.setState((prevState) => ({
-      cartArray: prevState.cartArray.filter((product) => product.id !== id)
-    }), () => {
-      console.log(this.state.cartArray)
-    });
-  }
+
+  handleClick = (e, id, name, price) => {
```

Comment:
Mi chodziło o to, aby taką funkcje zdefiniować poza komponentem `Product` i ją przekazywać do wnętrza przez `props` - może jednak zostać ;)

PS. Zwróć uwagę jak bardzo zwiększyła się czytelność tego komponentu :)